test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -303,38 +307,42 @@ func BenchmarkPerfScheduling(b *testing.B) {
 	}
 }
 
-func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
-	var mu sync.Mutex
-	var dataItems []DataItem
+func runWorkload(testCtx context.Context, b *testing.B, tc *testCase, w *workload) []DataItem {
+	workloadCtx, subtestCancel := context.WithCancel(testCtx)
```

Comment:
why do we need this intermediate context?

Reply:
We don't, done.